src/gitHub/GitHubBranchListStep.ts

```diff
@@ -0,0 +1,60 @@
+/*---------------------------------------------------------------------------------------------
+ *  Copyright (c) Microsoft Corporation. All rights reserved.
+ *  Licensed under the MIT License. See License.txt in the project root for license information.
+ *--------------------------------------------------------------------------------------------*/
+
+import { AzureWizardPromptStep, IAzureQuickPickItem, nonNullProp } from "@microsoft/vscode-azext-utils";
+import { QuickPickItem } from "vscode";
+import { loadMoreQp } from "../constants";
+import { localize } from "../utils/localize";
+import { IGitHubContext } from "./IGitHubContext";
+import { Branches, GetBranchesParams, getBranches } from "./getBranches";
+
+export class GitHubBranchListStep extends AzureWizardPromptStep<IGitHubContext> {
+    picks: IAzureQuickPickItem<string>[];
+
+    public async prompt(context: IGitHubContext): Promise<void> {
+        // We always want fresh picks before prompting in case the user has pressed the back button
```

Comment:
If you go back and choose a different repository, you'll want branches from the new repository not the old one 🙂

Reply:
We could maybe conditionally check to see if the previous values have changed... let me know if you prefer that logic instead... 🤔

Reply:
Oh, actually that makes sense. I was thinking more in a repository context, it'd be good not to remove all of the old picks, but honestly it's probably more trouble than it's worth.

For example, it could be useful if they only go back to repo, and then selected a new one from there, but if they back all the way up to org, then we'd have to clear picks if they picked a different org.

It's probably safer to just clear the picks as you had it.